### PR TITLE
convert numbers properly, so they're no longer treated as strings (go…

### PR DESCRIPTION
**Pull request by pierrestroh**
…t concatenated previously)

### Summary
adapted javascript calculations

### Motivation
dashboard displayed misleading/wrong data in graphs

### CODE REVIEW

**Comment by kraih** [comment]
Please elaborate. That description says practically nothing at the moment.

File: lib/Mojolicious/Plugin/Minion/resources/templates/minion/dashboard.html.ep [hunk 1]
```diff
@@ -40,7 +40,7 @@
         for (hour = 0; hour < data.daily.length; hour++) { 
           historylog.push({
             x: new Date(data.daily[hour].epoch * 1000).getTime(),
-            y: data.daily[hour].finished_jobs + data.daily[hour].failed_jobs
+            y: parseInt(data.daily[hour].finished_jobs, 10) + parseInt(data.daily[hour].failed_jobs, 10)
```
Comment (kraih) [inline lib/Mojolicious/Plugin/Minion/resources/templates/minion/dashboard.html.ep]: All these lines look way too long.

**Comment by kraih** [comment]
Also, why would we want handle JSON data from Ajax requests inconsistently? You did not change any other parts of the code.

**Comment by CandyAngel** [comment]
This looks like the issue I had where DBI/Postgres returns strings instead of numbers with 32bit Perl (or 32bit integers in Perl) because they are bigint.

**Comment by pierrestroh** [comment]
The backend is sending all values as text:

`{"active_jobs":"1","active_locks":"0","active_workers":"1","delayed_jobs":"0","enqueued_jobs":"907171","failed_jobs":"3","finished_jobs":"6876","inactive_jobs":"0","inactive_workers":0,"uptime":"747898.969006"}`

So for example:

`y: data.daily[hour].finished_jobs + data.daily[hour].failed_jobs`

When finished_jobs is "180" and failed_jobs is "0", y becomes "1800" (because they are treated as strings). By casting them explicitly, we ensure that the calculation are performed in the correct way instead of concatenating the values as string.

About the formatting: see other commit.

**Comment by CandyAngel** [comment]
For reference, this was discussed in IRC 2018-04-13, which is when I came across the same issue.

File: lib/Mojolicious/Plugin/Minion/resources/templates/minion/dashboard.html.ep [hunk 2]
```diff
@@ -37,10 +41,11 @@
       $.get('<%= url_for 'minion_history' %>').done(function (data) {
         var hour;
         var historylog = [];
-        for (hour = 0; hour < data.daily.length; hour++) { 
+        for (hour = 0; hour < data.daily.length; hour++) {
           historylog.push({
             x: new Date(data.daily[hour].epoch * 1000).getTime(),
-            y: data.daily[hour].finished_jobs + data.daily[hour].failed_jobs
+            y: parseInt(data.daily[hour].finished_jobs, 10) +
+               parseInt(data.daily[hour].failed_jobs, 10)
```
Comment (kraih) [inline lib/Mojolicious/Plugin/Minion/resources/templates/minion/dashboard.html.ep]: Indentation is inconsistent.

**Comment by kraih** [comment]
And i cannot confirm your findings, with the default Pg backend i only see numbers in the JSON generated by `/minion/stats`.

**Comment by pierrestroh** [comment]
When we Dump one element of $daily in `Minion::Backend::Pg` via `Devel::Peek` before returning it:

```
SV = IV(0x4ce4298) at 0x4ce42a8
  REFCNT = 1
  FLAGS = (ROK)
  RV = 0x4ce4188
  SV = PVHV(0x4b707c0) at 0x4ce4188
    REFCNT = 1
    FLAGS = (SHAREKEYS,HASKFLAGS)
    ARRAY = 0x4cd13a0  (0:6, 1:1, 2:1)
    hash quality = 90.0%
    KEYS = 3
    FILL = 2
    MAX = 7
    Elt "epoch" [UTF8 "epoch"] HASH = 0x5c12e5fa
    SV = PV(0x4cb5ce0) at 0x4ce4308
      REFCNT = 1
      FLAGS = (POK,IsCOW,pPOK,UTF8)
      PV = 0x4d08fa0 "1529415803.78888"\0 [UTF8 "1529415803.78888"]
      CUR = 16
      LEN = 18
      COW_REFCNT = 0
    Elt "finished_jobs" [UTF8 "finished_jobs"] HASH = 0x96af38e2
    SV = PV(0x4cb5cd0) at 0x4ce42f0
      REFCNT = 1
      FLAGS = (POK,pPOK,UTF8)
      PV = 0x4cd13f0 "290"\0 [UTF8 "290"]
      CUR = 3
      LEN = 10
    Elt "failed_jobs" [UTF8 "failed_jobs"] HASH = 0xa4cd0a04
    SV = PV(0x4cb5cc0) at 0x4ce42d8
      REFCNT = 1
      FLAGS = (POK,pPOK,UTF8)
      PV = 0x4cd1380 "0"\0 [UTF8 "0"]
      CUR = 1
      LEN = 10
```
So it is returned as string.

PostgreSQL 9.6.9
Perl 5.26.1 64bit
Mojolicious 7.84
Minion 9.03
Mojo::Pg 4.08
DBI 1.631
DBD::Pg 3.5.1

**Comment by pierrestroh** [comment]
Note: upgrading DBD::Pg to latest (DBD-Pg-3.7.4) fixed the issue.

Guarding against older versions of DBD::Pg would be nice though?

**Comment by Grinnz** [comment]
It is a good idea to not have the javascript rely on whether the database or the JSON encoder decides that something is a number or not, IMO. DBD::mysql for instance only returns strings.

**Comment by kraih** [comment]
Looks like this pull request went stale, therefore closing it for now.

**Pull request closed without merging**